python/nano/src/bigdl/nano/pytorch/torch_nano.py

```diff
@@ -36,17 +37,31 @@
 
 class _TorchNanoModule(_LiteModule):
     def state_dict(self, *args, **kwargs):
-        return self._module.state_dict(*args, **kwargs)
+        if isinstance(self.module, DistributedDataParallel):
+            return self.module.module.state_dict(*args, **kwargs)
+        else:
+            return self.module.state_dict(*args, **kwargs)
 
     def load_state_dict(self, *args, **kwargs):
         invalidInputError(False, "TorchNano doesn't support loading state dict, "
                           "please load it using original pytorch model")
 
     def __getattr__(self, name: str):
-        # automatically unwrap attributes access of _LiteModule
+        # automatically unwrap attributes access of _LiteModule,
+        # always throw a single-level exception when the attribute doesn't exist
+        # for a more user-friendly exception message
         try:
             return super().__getattr__(name)
         except AttributeError:
+            pass
+
+        if isinstance(self.module, DistributedDataParallel):
+            try:
```

Comment:
maybe adding some comments on why trying to access `self.module.custome_attr` first?

Reply:
done